mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -226,7 +227,7 @@ impl State for ValidateTakerPayment {
             )
             .compat();
         if let Err(err) = wait_fut.await {
-            Self::change_state(Stopped::from_reason(StopReason::Error(
+            return Self::change_state(Stopped::from_reason(StopReason::Error(
```

Comment:
Sorry, just curious. Why didnt you use return in your previous PR? Is it because you tested the implementation?

Reply:
No this was a bug :)